### PR TITLE
Remove derivative of specific heat from NS kernels

### PR DESCRIPTION
**Pull request by GiudGiud**
If h = cp T with cp constant it doesnt matter
For our future formulation of h = int(cp dT) then it ll matter

Refs #21245

### CODE REVIEW

File: modules/navier_stokes/doc/content/source/fvkernels/PINSFVEnergyTimeDerivative.md [hunk 1]
```diff
@@ -13,8 +13,7 @@ for the fluid phase and
 \end{equation}
 for the solid phase, where $\epsilon$ is the porosity, $\rho_{f/s}$ the fluid/solid material density, $c_{pf/s}$ the fluid/solid specific heat and $T_{f/s}$ the fluid/solid temperature.
 
-The time derivatives of density and the specific heat capacity are ignored if [!param](/FVKernels/PINSFVEnergyTimeDerivative/drho_dt)
-and [!param](/FVKernels/PINSFVEnergyTimeDerivative/dcp_dt) are not provided. For incompressible flows, the former should not be provided.
+The time derivative of the density is ignored if [!param](/FVKernels/PINSFVEnergyTimeDerivative/drho_dt) is not provided. For incompressible flows, the former should not be provided.
```
Comment (grmnptr) [inline modules/navier_stokes/doc/content/source/fvkernels/PINSFVEnergyTimeDerivative.md:16]: I think we should include the derivation of the correct form in the description before this. I can append it to this PR if needed.

**Review by grmnptr — CHANGES_REQUESTED** [review]

File: modules/navier_stokes/doc/content/source/fvkernels/PINSFVEnergyTimeDerivative.md (hunk 1 above)
Comment (GiudGiud) [reply modules/navier_stokes/doc/content/source/fvkernels/PINSFVEnergyTimeDerivative.md:16]: added!

**Comment by GiudGiud** [comment]
We should probably get things in order for enthalpy vs energy too at some point.

File: modules/navier_stokes/doc/content/source/fvkernels/INSFVEnergyTimeDerivative.md [hunk 2]
```diff
@@ -5,14 +5,34 @@
...
+capacities being equal for incompressible fluids,
+
+\begin{equation}
+U(T(t), p(t)) = \rho(T(t), p(t)) \int_0^T(t) dT cp(T(t), p(t))
+\end{equation}
+
+we take the partial derivative with regards to time:
+
+\begin{equation}
+\frac{\partial U}{\partial t}} = \frac{\partial \rho(T, p)}{\partial t}} \int_0^T dT cp(T, p) + \rho(T, p) \frac{\partial T}{\partial t}} cp(T, p)
```
Comment (grmnptr) [inline modules/navier_stokes/doc/content/source/fvkernels/INSFVEnergyTimeDerivative.md:28]: This does not parse well.
```suggestion
\frac{\partial U}{\partial t}} = \frac{\partial \rho(T, p)}{\partial t} \int_0^T dT cp(T, p) + \rho(T, p) \frac{\partial T}{\partial t} cp(T, p)
```

**Review by grmnptr — CHANGES_REQUESTED** [review]
The equation in the derivation doesn't compile.

**Comment by GiudGiud** [comment]
should be good now

File: modules/navier_stokes/doc/content/source/fvkernels/INSFVEnergyTimeDerivative.md [hunk 3]
```diff
@@ -5,14 +5,34 @@
...
+capacities being equal for incompressible fluids,
+
+\begin{equation}
+U(T(t), p(t)) = \rho(T(t), p(t)) \int_0^T(t) dT cp(T(t), p(t))
+\end{equation}
+
+we take the partial derivative with regards to time:
+
+\begin{equation}
+\frac{\partial U}{\partial t}} = \frac{\partial \rho(T, p)}{\partial t} \int_0^T dT cp(T, p) + \rho(T, p) \frac{\partial T}{\partial t} cp(T, p)
```
Comment (grmnptr) [inline modules/navier_stokes/doc/content/source/fvkernels/INSFVEnergyTimeDerivative.md:28]: One more, maybe you can squash this guy.
```suggestion
\frac{\partial U}{\partial t} = \frac{\partial \rho(T, p)}{\partial t} \int_0^T dT cp(T, p) + \rho(T, p) \frac{\partial T}{\partial t} cp(T, p)
```

**Review by grmnptr — CHANGES_REQUESTED** [review]

**Comment by GiudGiud** [comment]
done. I would build docs locally but all CPUs are on NONUNITY build

**Review by grmnptr — APPROVED** [review]

**Pull request merged**